packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -191,8 +188,8 @@ export class OdspDocumentService implements IDocumentService {
         return getWithRetryForTokenRefresh<IDocumentDeltaConnection>(async (options) => {
             // For ODC, we just use the token from joinsession
             const socketTokenPromise = this.isOdc ? Promise.resolve("") : this.getWebsocketToken(options);
-            const [websocketEndpoint, webSocketToken, io] =
-                await Promise.all([this.joinSession(), socketTokenPromise, this.socketIoClientFactory()]);
```

Comment:
From perf POV, that triple fetch is the best outcome as all of them happen in parallel (well, in practice, it's only web socket tone & socket io bundle). Moving things around will regress perf, as we would have to do things sequntially.